src/qibo/models.py

```diff
@@ -429,3 +431,161 @@ def _loss(params, qaoa, hamiltonian):
                                                              compile=compile, processes=processes)
         self.set_parameters(parameters)
         return result, parameters, extra
+
+
+class Grover:
+    """Model that performs Grover's algorithm.
+
+    # TODO: Add a few more details on Grover's algorithm and/or reference.
+
+    Args:
+        oracle (:class:`qibo.core.circuit.Circuit`): quantum circuit that flips
+            the sign using a Grover ancilla initialized with -X-H-
+            and expected to have the total size of the circuit.
+        initial_state (:class:`qibo.core.circuit.Circuit`): quantum circuit
+            that initializes the state. Leave empty if |000..00>
+        superposition (:class:`qibo.core.circuit.Circuit`): quantum circuit that
```

Comment:
Maybe we should rename it `superposition_circuit` in order to make it transparent.